### DISCUSSION TITLE
module 'llama_cpp' has no attribute 'llama_init_from_file'

### DISCUSSION THREAD

**Discussion by Vadim170** [abetlen/llama-cpp-python#673] [Q&A]
Hello, please help me figure out what is the cause of the error with low level api. Hight level api works fine

My steps:
git clone git@github.com:abetlen/llama-cpp-python.git
cd llama-cpp-python
git submodule update --init —recursive
python3 setup.py develop
…
Finished processing dependencies for llama-cpp-python==0.1.83

And I get this error:
llama-cpp-python > python3 examples/low_level_api/low_level_api_chat_cpp.py -m pathto/ggml-model-Q4_0.bin
seed = 1693947851
Traceback (most recent call last):
  File "/Users/vadimmakarov/Documents/Work_home/llama/llama-cpp-python/examples/low_level_api/low_level_api_chat_cpp.py", line 567, in <module>
    with LLaMAInteract(params) as m:
  File "/Users/vadimmakarov/Documents/Work_home/llama/llama-cpp-python/examples/low_level_api/low_level_api_chat_cpp.py", line 69, in __init__
    self.ctx = llama_cpp.llama_init_from_file(self.params.model.encode("utf8"), self.lparams)
AttributeError: module 'llama_cpp' has no attribute 'llama_init_from_file'

P.S.
directory state:
llama-cpp-python > git show --summary
commit 18337267c175f88c46fbad079a7c18da57e1b520 (HEAD -> main, origin/main, origin/HEAD)
Merge: 186626d bf08d1b
Author: Andrei Betlen <abetlen@gmail.com>
Date:   Fri Sep 1 14:26:16 2023 -0400

    Merge branch 'main' of https://github.com/abetlen/llama-cpp-python into main

llama-cpp-python > git submodule status
 69fdbb9abc8907dd2a9ffdd840cba92d678a660a vendor/llama.cpp (b1147-1-g69fdbb9)

**Comment by N0THSA** [comment]
I'm having this exact same issue. My solution was to downgrade to 0.1.73 at the cost of GGUF models.

I'm currently running it using the "examples/low/examples/low_level_api" folder running "python3 Chat.py"

Output:

```seed = 1371195711
Traceback (most recent call last):
  File "/home/nothsa/Music/term/start.py", line 78, in <module>
    with LLaMAInteract(params) as m:
         ^^^^^^^^^^^^^^^^^^^^^
  File "/home/nothsa/Music/term/low_level_api_chat_cpp.py", line 62, in __init__
    self.ctx = llama_cpp.llama_init_from_file(self.params.model.encode("utf8"), self.lparams)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'llama_cpp' has no attribute 'llama_init_from_file'
[nothsa@nothsa-terminal term]$ 
```
This occurred right before it would normally load the model.

Edit: This issue seems to be affecting at least GGUF AND GGML models.

**Comment by N0THSA** [comment]
#680 This pull request should be the fix. Should get merged anytime.
